src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs

```diff
@@ -62,14 +63,18 @@ let private runner (args: AstNodeRuleParams) =
             | SynExpr.Const (SynConst.String (id, _, _), _) when id = "" ->
                 generateError failwithId.idText id range BadUsageType.EmptyMessage maybeIdentifier
             | SynExpr.Const (SynConst.String (id, _, _), _) ->
-                if Set.contains id failwithErrorMessageList then
+                let testRange =
+                    not (List.isEmpty failwithRangeList)
+                    && List.exists (fun (ident, rng) -> id = ident && not (range = rng)) failwithRangeList
+
+                if testRange then
                     generateError failwithId.idText id range BadUsageType.DuplicateMessage maybeIdentifier
                 else
                     match maybeIdentifier with
                     | Some maybeId ->
                         generateError failwithId.idText id range BadUsageType.SwallowedException (Some maybeId)
                     | _ ->
-                        failwithErrorMessageList <- failwithErrorMessageList.Add(id)
+                        failwithRangeList <- List.append (List.singleton (id, range)) failwithRangeList
```

Comment:
AFAIU you don't need to use append here, you can just concatenate with `::`

Reply:
Done

Reply:
@janus you shouldn't say "Done" until you push, I don't see the addressing of this issue in the PR.